### PR TITLE
Make csrf support optional

### PR DESCRIPTION
**Pull request by olOwOlo**
#### What's this PR do/fix?

Make csrf support optional.

#### Are there unit tests? If not how should this be manually tested?

Update existing unit tests.

#### Any background context you want to provide?

None.

#### What are the relevant issues?

#2578

### CODE REVIEW

**Comment by dilipkrish** [comment]
Thanks @olOwOlo!! Will pull it in shortly

File: springfox-swagger-ui/src/web/js/csrf.js [hunk 1]
```diff
@@ -4,6 +4,8 @@
  * @returns {Promise<void>}
  */
 export default async function patchRequestInterceptor(baseUrl) {
+  if (!window.ui.findCsrfToken) return;
```
Comment (fennekit) [inline springfox-swagger-ui/src/web/js/csrf.js]: Are the configuration values not accessed via  window.ui.getConfigs().findCsrfToken ??

File: springfox-swagger-ui/src/web/js/springfox.js [hunk 2]
```diff
@@ -43,6 +43,7 @@ window.onload = () => {
       });
 
       window.ui = getUI(baseUrl, resources, configUI, configSecurity);
+      window.ui.findCsrfToken = configSecurity.findCsrfToken;
```
Comment (fennekit) [inline springfox-swagger-ui/src/web/js/springfox.js]: different access pattern than the rest of the javascript configuration options

**Comment by Aloren** [comment]
@olOwOlo any chance you could update the pull request? thanks

**Comment by hokkun-dayo** [comment]
@olOwOlo 
I'd appreciate it if you could update this PR :)

**Comment by olOwOlo** [comment]
Sorry for the delay. 
Thank you @fennekit for pointing out the problem. After some investigation, I found that there is a `custom` parameter here (https://github.com/swagger-api/swagger-ui/pull/3108/commits/b530ac8593c292a24f15d015b5246d517428854f). So, I moved this configuration options to here.

**Comment by matthewh86** [comment]
Hi @dilipkrish, is this PR going to be merged? It's causing some minor (but still annoying) issues with the login pop-up appearing because of Spring security / Basic Auth. I can configure a WebSecurityConfigurerAdapter class to permit the "/" and "/csrf" endpoints, but this seems hacky.

```java
@Configuration
@EnableWebSecurity
public class WebSecurityConfiguration extends WebSecurityConfigurerAdapter {

    @Override
    protected void configure(final HttpSecurity http) throws Exception {

        http.csrf().disable()
            .sessionManagement().sessionCreationPolicy(STATELESS)
            .and()
            .authorizeRequests()
            .requestMatchers(to(HealthEndpoint.class, InfoEndpoint.class, PrometheusScrapeEndpoint.class)).permitAll()
            .antMatchers("/api-docs", "/swagger-ui.html", "/webjars/**", "/swagger-resources/**").permitAll()
            .antMatchers("/", "/csrf").permitAll() // hacky
            .antMatchers("/**").authenticated()
            .and().httpBasic();
    }
}
```

**Comment by dilipkrish** [comment]
Thank you for the PR @olOwOlo! Sorry it took so long to get this in.

**Pull request merged**